### ISSUE TITLE
False positive `PD008` when using `zipfile.Path.at`

### ISSUE THREAD

**Issue by m-roberts** [charliermarsh/ruff#4466]
`ruff==0.0.267` has a false positive `PD008`:

```python
import io
import zipfile


class MockBinaryFile(io.BytesIO):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def close(self):
        pass  # Don't allow closing the file, it would clear the buffer


zip_buffer = MockBinaryFile()

with zipfile.ZipFile(zip_buffer, "w") as zf:
    zf.writestr("dir/file.txt", "This is a test")

# Reset the BytesIO object's cursor to the start.
zip_buffer.seek(0)

with zipfile.ZipFile(zip_buffer, "r") as zf:
    zpath = zipfile.Path(zf, "/")

dir_name, file_name = zpath.at.split("/")
```

```
> ruff --select=PD008 a.py
a.py:24:23: PD008 Use `.loc` instead of `.at`.  If speed is important, use numpy.
   |
24 | dir_name, file_name = zpath.at.split("/")
   |                       ^^^^^^^^ PD008
   |

Found 1 error.
```

**Comment by charliermarsh** [comment]
Thanks! Some of the `PD` false positives are hard to avoid, but this one should be fixable.
